src/webextension/survey/components/app.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Uri from 'urijs';
+import uuid from 'uuid';
+
+import { sendMessage } from '../../lib/util';
+import Survey from './survey.jsx';
+
+class App extends React.Component {
+  // Generate and cache initial form values.
+  getInitialValues() {
+    // If we've already calculated this, just return them.
+    if (this.initialValues) {
+      return this.initialValues;
+    }
+
+    // Ensure that optional fields are included in the payload, even if empty.
+    const defaultValues = { details: '' };
+
+    // Generate a UUID, then send to the host SDK add-on so that it can
+    // associate a future submission with the correct tab and window.
+    defaultValues.id = uuid();
+    sendMessage('id', defaultValues.id);
+
+    // Get `sentiment` and `type` from the querystring.
+    var qs = new Uri(window.location.search).search(true);
```

Comment:
Not sure about your religious feelings re: `var` vs `let`/`const`.

If you prefer `let` and `const` exclusively, we can add the following rules to our ESLint config:

```yaml
    "no-var": "error",
    "prefer-const": "error",
```